src/renderer/viz/expressions/top.js

```diff
@@ -62,53 +67,59 @@ export default class Top extends BaseExpression {
         return Math.round(this.buckets.eval()) + 1;
     }
     _applyToShaderSource (getGLSLforProperty) {
-        const property = this.property._applyToShaderSource(getGLSLforProperty);
+        const childSources = {};
+        this.childrenNames.forEach(name => { childSources[name] = this[name]._applyToShaderSource(getGLSLforProperty); });
         return {
-            preface: this._prefaceCode(property.preface + `uniform sampler2D topMap${this._uid};\n`),
-            inline: `(255.*texture2D(topMap${this._uid}, vec2(${property.inline}/1024., 0.5)).a)`
+            preface: this._prefaceCode(Object.values(childSources).map(s => s.preface).reduce((a, b) => a + b, '') + `
+            float top${this._uid}(float id){
```

Comment:
OK, so there are 8 buckets based on the feature id?

Reply:
There is a maximum of 8 buckets for `top` `top($wadus, 9)` doesn't work now. I expect to increase this to `16`